solana/rpc/websocket.py

```diff
@@ -0,0 +1,162 @@
+"""API WebSocket Client to interact with the Solana JSON RPC Endpoint."""
+from __future__ import annotations
+
+from typing import Union
+
+from solana.publickey import PublicKey
+
+from .providers.ws import Provider
+from .types import RPCMethod, RPCResponse
+
+
+class WebSocketClient:
+    """Websocket client class."""
+
+    def __init__(self, endpoint: str):
+        """Init API client for websocket."""
+        self._provider = Provider(endpoint)
+
+    def account_subscribe(self, pubkey: Union[PublicKey, str]) -> RPCResponse:
+        """Subscribe to an account to receive when lamports change.
+
+        param pubkey: Pubkey of account to receive lamports, as base-58 encoded string or public key object.
+
+        >>> from solana.publickey import PublicKey #doctest: +SKIP
+        >>> ws_client = WebSocketClient("ws://localhost:8900") #doctest: +SKIP
+        >>> ws_client.account_subscribe(PublicKey(1)) # doctest: +SKIP
+        {'jsonrpc': '2.0', 'result': 1, 'id': 1}
+        """
+        return self._provider.make_request(RPCMethod("accountSubscribe"), str(pubkey))
+
+    def account_unsubscribe(self, account_id: int) -> RPCResponse:
+        """Unsubscribe from account change notifications.
+
+        param account_id: id of account Subscription to cancel
+
+        >>> ws_client = WebSocketClient("ws://localhost:8900") #doctest: +SKIP
+        >>> resp = ws_client.account_subscribe(PublicKey(1)) #doctest: +SKIP
+        >>> account_id = resp['result'] #doctest: +SKIP
+        >>> ws_client.account_unsubscribe(account_id) #doctest: +SKIP
+        {'jsonrpc': '2.0', 'result': True, 'id': 3}
+        """
+        return self._provider.make_request(RPCMethod("accountUnsubscribe"), account_id)
+
+    def program_subscribe(self, pubkey: Union[PublicKey, str]) -> RPCResponse:
```

Comment:
Handle options for `encoding` and `commitment`.

https://docs.solana.com/apps/jsonrpc-api#programsubscribe